lookup.go

```diff
@@ -121,5 +121,5 @@ func (dht *IpfsDHT) GetClosestPeers(ctx context.Context, key string) (<-chan pee
 		dht.routingTable.ResetCplRefreshedAtForID(kadID, time.Now())
 	}
 
-	return out, nil
+	return out, ctx.Err()
```

Comment:
Do we want to return out or nil if ctx.Err != nil? If we want to return out then we need a function documentation comment since by default people tend to do "if err != nil {return err}; continue assuming normal operation". Letting people know that useful info is returned even when we error is required.

Reply:
We should definitely document this. My main concern is that I want to tell users that we didn't _actually_ reach the end of the query, we stopped early. However, the user _could_ decide to use the partial result anyways.

I guess this answers https://github.com/libp2p/go-libp2p-kad-dht/pull/478#discussion_r388492203. We can still get partial results and might as well use them. However, if we do that, we still need to make sure to return an error to the user.